plugins/woocommerce/includes/admin/helper/class-wc-helper-subscriptions-api.php

```diff
@@ -56,6 +56,12 @@ public static function register_rest_routes() {
 				'methods'             => 'POST',
 				'callback'            => array( __CLASS__, 'activate' ),
 				'permission_callback' => array( __CLASS__, 'get_permission' ),
+				'args'                => array(
+					'product_key' => array(
+						'required' => true,
+						'type'     => 'string',
+					),
+				),
```

Comment:
Likely not, you are right. Free products without product keys won't show up on the table _now_, but as I understand it @bgrgicak will change this. So this might be a change for a future PR.

Reply:
Nice, thanks, just been reviewing Bero's [PR](https://github.com/woocommerce/woocommerce/pull/40820#pullrequestreview-1684536781) for this.